cpp/src/cider/exec/processor/DefaultBatchProcessor.cpp

```diff
@@ -91,7 +91,7 @@ void DefaultBatchProcessor::processNextBatch(const struct ArrowArray* array,
     CIDER_THROW(CiderRuntimeException,
                 getErrorMessageFromErrCode(static_cast<cider::jitlib::ERROR_CODE>(ret)));
   }
-
+  runtime_context_->destroyStringHeap();
```

Comment:
may be expose a more generic method named `destroy` and  invoke `destroyStringHeap` inside runtimeContext?

Reply:
`destroyStringHeap` will be called for each `processNextBatch` call. while `destroy` should be called after the last `processNextBatch` call.